src/components/Table/TableActions.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { Material } from '@balena/ui-shared-components';
+import type { AutoUIEntityPropertyDefinition } from '../../AutoUI';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCog } from '@fortawesome/free-solid-svg-icons/faCog';
+
+const {
+	IconButton,
+	Menu,
+	MenuItem,
+	Checkbox,
+	FormGroup,
+	FormControlLabel,
+	Divider,
+} = Material;
+
+interface TableActionsProps<T> {
+	columns: Array<AutoUIEntityPropertyDefinition<T>>;
+	actions?: Material.MenuItemProps[];
+	onColumnPreferencesChange?: (
+		columns: Array<AutoUIEntityPropertyDefinition<T>>,
+	) => void;
+}
+
+export const TableActions = <T extends object>({
+	columns,
+	actions,
+	onColumnPreferencesChange,
+}: TableActionsProps<T>) => {
+	const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+	const open = Boolean(anchorEl);
+	const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+		setAnchorEl(event.currentTarget);
+	};
+	const handleClose = () => {
+		setAnchorEl(null);
+	};
+	const handleColumnSelection = React.useCallback(
+		(column: AutoUIEntityPropertyDefinition<T>) => {
+			if (!onColumnPreferencesChange) {
+				return;
+			}
+			if (typeof column.label === 'string' && column.label.startsWith('Tag:')) {
+				onColumnPreferencesChange(columns.filter((c) => c.key !== column.key));
+				return;
+			}
+			const newColumns = columns.map((c) =>
```

Comment:
A `set` may be more fitting for `columns` 🤔 In this case we would update specifically the column with the given key. If possible and not too difficult, I would see if changing `columns` to a set is possible. But if not, it's okay, would just be nice to know what the blocker is if any

Reply:
A Set is beneficial if you frequently need to check the existance of a value or perform operations like union, intersection, or difference.
newColumns is derived through a map operation, which does not benefit from the properties of a Set.
If you mean to generally change columns in general into a set then we can think to it, but I would probably do it in a different PR.

Reply:
Yep I meant columns in general. We can think about it for a separate PR